http-clients/url-connection-client/src/main/java/software/amazon/awssdk/http/urlconnection/UrlConnectionHttpClient.java

```diff
@@ -159,10 +162,18 @@ public void abort() {
          * means infinity, and is not recommended.
          */
         Builder connectionTimeout(Duration connectionTimeout);
+
+        /**
+         * A function used to create a {@link HttpURLConnection} from a {@link URL}.
+         *
+         * By default {@link URL#openConnection()} is used.
+         */
+        Builder connectionCreator(Function<URL, HttpURLConnection> connectionCreator);
```

Comment:
1. Should we use our own factory interface, instead of a built-in, so that we can document it better?
2. Do we override customer settings with our own? 
3. If so, how do we handle new, SDK-provided settings in the future? They could potentially override customer settings and break backwards-compatibility.